### PR TITLE
doc: add a section about coverity scans

### PR DESCRIPTION
**Pull request by nashif**
Quick intro into static code analysis and scans.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: doc/contribute/guidelines.rst [hunk 1]
```diff
@@ -506,6 +506,51 @@ issues, you can add option --no-verify to the git push command.
...
+
+Known limitations
+=================
+
+Placeholder for list of known issues and limitations of the scan service.
+
+Workflow
+========
+
+If the problem is listed under Known limitations then please set the
```
Comment (carlescufi) [inline doc/contribute/guidelines.rst:544]: what are "Known limitations" in Zephyr?

File: doc/contribute/guidelines.rst [hunk 2]
```diff
@@ -506,6 +506,51 @@ issues, you can add option --no-verify to the git push command.
...
+classification to either "False positive" or "Intentional", the action to
+"Ignore", owner to your own account and add a comment why the issue is
+considered false positive or intentional.
+
+Update the related issue in the Zephyr project with the details and only close
+the issue after completing the step above on scan service website. Any issues
+closed without a fix or without ignore the entry in the scan service will be
```
Comment (carlescufi) [inline doc/contribute/guidelines.rst:551]:
```suggestion
closed without a fix or without ignoring the entry in the scan service will be
```

File: doc/contribute/guidelines.rst (hunk 1 above)
Comment (carlescufi) [reply doc/contribute/guidelines.rst:544]: shouldn't we say "If, after analyzing the Coverity report, you come to the conclusion that it is a false positive, ..." ?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: doc/contribute/guidelines.rst [hunk 3]
```diff
@@ -506,6 +506,51 @@ issues, you can add option --no-verify to the git push command.
...
+abstract interpretation to gain information about the code's control flow and
+data flow. It's able to follow all possible code paths that a program may take.
+For example the analyzer understands that malloc() returns a memory that must
+be freed with free() later. It follows all branches and function calls to see
+if all possible combinations free the memory. The analyzer is able to detect
+all sorts of issues like resource leaks (memory, file descriptors), NULL
+dereferencing, use after free, unchecked return values, dead code, buffer
+overflows, integer overflows, uninitialized variables, and many more.
+
+The results are available on the `Coverity Scan <https://scan.coverity.com/>`_
```
Comment (kartben) [inline doc/contribute/guidelines.rst:527]: The project URL is public so prob better to link to it directly?

```suggestion
The results are available on the `Coverity Scan <https://scan.coverity.com/projects/zephyr>`_
```

File: doc/contribute/guidelines.rst [hunk 4]
```diff
@@ -506,6 +506,51 @@ issues, you can add option --no-verify to the git push command.
...
+website. In order to access the results you have to create an account yourself.
+Then go to Projects using Scan and add yourself to the Zephyr project. New
```
Comment (kartben) [inline doc/contribute/guidelines.rst:529]: can be reworded if link above is changed to a direct link to the Zephyr scan. 

```suggestion
From the Zephyr project page, you may select "Add me to project" to be added to the project. New
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/contribute/guidelines.rst (hunk 1 above)
Comment (nashif) [reply doc/contribute/guidelines.rst:544]: > what are "Known limitations" in Zephyr?

Is is not about zephyr, it is limiation of tool itself. I added a section above which will be populated. For example, coverity, it seems, has an assue with zephyr ASSERTs which might result in lots of false positives. We do not check for errors and rely on ASSERTs,  and the tool thinks it is a problem.

Comment (nashif) [reply doc/contribute/guidelines.rst:544]: > what are "Known limitations" in Zephyr?

Is is not about zephyr, it is limiation of tool itself. I added a section above which will be populated. For example, coverity, it seems, has an assue with zephyr ASSERTs which might result in lots of false positives. We do not check for errors and rely on ASSERTs,  and the tool thinks it is a problem.

**Review by kartben — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**